requirements.txt

```diff
@@ -1,27 +1,12 @@
-astroid==1.5.3
-backports.functools-lru-cache==1.4
-boto==2.34.0
-colorama==0.3.9
-enum34==1.1.6
-Flask==0.12.4
-Flask-Cors==3.0.3
-Flask-HTTPAuth==3.2.3
-Flask-Session==0.3.1
-Flask-SSLify==0.1.5
-funcsigs==1.0.2
-itsdangerous==0.24
-Jinja2==2.10.1
-lazy-object-proxy==1.3.1
-MarkupSafe==1.0
-mock==1.3.0
-pbr==3.1.1
-psycopg2==2.7.3.2
-py==1.4.34
-pylint==1.5.2
-pytest==2.9.2
-requests==2.20.0
-singledispatch==3.4.0.3
-six==1.11.0
-uWSGI==2.0.15
-Werkzeug==0.12.2
-wrapt==1.10.11
+autopep8
```

Comment:
these all should have at least a major version constraint, or updating the lock with newly released versions will be very difficult?

Reply:
I'll look into it

> or updating the lock with newly released versions will be very difficult?

I guess the point is that locking/pinning the requirements only happens occasionally and when it does happen then it's part of an upgrade process. In some cases there are conflicts among dependencies that require specifying version ranges or conditions that should be captured in requirements.txt

on the other hand when there is some version bump that needs to happen because of a security flaw, I just want to face-roll it and tightening up the requirements.txt file could help there

Reply:
this was interesting: https://www.python.org/dev/peps/pep-0440/#compatible-release